week-2/mandatory/3-api/cyf_ecommerce_api/server.js

```diff
@@ -0,0 +1,215 @@
+        db.query(query,[NewProduct],(err)=>{
+        if(err==undefined){
+            res.send("Product created."); 
+        }else{
+            res.status(500).json({error: err});
+        }
+    })
+})
+
+// Add a new POST endpoint `/availability` to create a new product availability (with a price and a supplier id). Check that the price is a positive integer and that both the product and supplier ID's exist in the database, otherwise return an error.
+app.post("/availability/:prod_id",(req,res)=>{
+    const prod_id=parseInt(req.params.prod_id)
+    const NewPrice=req.body.unit_price
+    const NewSupplierId=req.body.supp_id
+const query="INSERT INTO product_availability (prod_id,unit_price,supp_id) VALUES ($1,$2,$3) "
+if(NewPrice>0){
+db.query(query,[prod_id,NewPrice,NewSupplierId],(err)=>{
+    if(err==undefined){
+        res.send("New price updated")
+    }else{
+        res.status(500).json({error: err});
+    }})
+}else{
+    res.send("Please enter a valid Price")
+}})
+
+
+// Add a new POST endpoint `/customers/:customerId/orders` to create a new order (including an order date, and an order reference) for a customer. Check that the customerId corresponds to an existing customer or return an error.
+app.post("/customers/:customerId/orders",(req,res)=>{
+const customer_id=parseInt(req.params.customer_id)
+const reference=req.body.order_reference
+const query="INSERT INTO orders (order_date,order_reference,customer_id) VALUES (current_date,$1,$2,)"
+db.query(query,[reference,customer_id],(err,result)=>{
+        if(err==undefined){
+            newCustomerId = result.rows[0].id;
+            res.status(200).json({newCustomerId:newCustomerId})
+        }else{
+            console.log(err);
+            res.status(500).json({error: err});
+        }
+})})
+
+// Add a new PUT endpoint `/customers/:customerId` to update an existing customer (name, address, city and country).
+app.put("/customers/:customerId",(req,res)=>{
+    const NewName=req.body.name
+    const NewAddress=req.body.address
+    const NewCity=req.body.city
+    const NewCountry=req.body.country
+    db.query("UPDATE customers SET name=$1,address=$2,city=$3,country=$4",[NewName,NewAddress,NewCity,NewCountry],(err,result)=>{
+        if(err==undefined){
+            res.status(200).json("Customer Updated")
+        }else{
+            res.status(500).json({error:err})
+        }
+    })
+})
+
+//Add a new DELETE endpoint `/orders/:orderId` to delete an existing order along with all the associated order items.
+app.delete("/orders/:id",(req,res)=>{
+    const orderId=Number(req.params.id)
+    db.query("DELETE FROM order_items where order_id=$1",[orderId],(err,result)=>{
+        if(err==undefined){
+            if(result.rowCount==0){
+                res.status(400).send("There isn't any order with this ID.");
+            }else if(result.rowCount==1){
+                res.status(200).json("Order is deleted")
+            }
```

Comment:
What would be the 3rd case when more than 1 row has been deleted? I know it is not valid in this case because we use the primary key to delete an element. So delete will effect to either zero or one row.

Hence you use the primary key to delete a row, you don't really need to check whether 1 or more rows have been deleted unless there is a reason to do so.

But if you do check whether 1 row is deleted or not, you should then also handle the case for more than 1. Otherwise, the response will be never fulfilled, it will hang until being aborted.